### DISCUSSION TITLE
Cloudflare缓存可能导致mkv metadata缺失，从而导致视频不能拖动及其他问题

### DISCUSSION THREAD

**Discussion by quul** [AlistGo/alist#1128] [General]
## 事情经过
在buyvm的卢森堡小鸡上搭建了个alist用来看视频，由于卢森堡网络实在不太行就套了一层cf。

在实际观看的时候发现视频无法拖动（或是有的客户端拖动后需要缓存很长时间）便进行排查。

尝试使用mpv打开文件提示
```
➜  ~ mpv https://example.com/downloads/The.West.Wing.S01-S07.COMPLETE.1080p.AMZN.WEB-DL.DD%2B2.0.H.264-P2P/The.West.Wing.S07.1080p.AMZN.WEB-DL.DD%2B2.0.H.264-AJP69/The.West.Wing.S07E01.The.Ticket.1080p.AMZN.WEB-DL.DD%2B2.0.H.264-AJP69.mkv
[mkv] Stream is not seekable or unknown size, not reading mkv metadata at end of file.
[mkv] Stream is not seekable or unknown size, not reading mkv metadata at end of file.
```
经控制变量对比最终发现是cloudflare的缓存策略导致的，在裸连或打开Development mode时一切表现正常（或者直接开个caddy file_server也能复现问题）。

## 解决方案（
在 Cloudflare Page Rules 里直接设置不缓存mkv文件
```
*.example.com/*.mkv
Cache Level: Bypass
```
非常的简单粗暴
## ~其他吐槽~
不太熟悉cf的缓存规则以及mkv的结构，这是一个能用解决方案，搜了半天也没找到有其他人提到过这个问题。描述存在问题请大佬轻喷（

**Comment by aur3l14no** [comment]
Thanks for sharing!
